### PR TITLE
Fix mulebot movement while controlled by a player

### LINKED ISSUE #52281: Mule power bug and Pai Mule speed and power bug
## Reproduction:
-Mule and characters with diagnostic glasses don't have power hud on Mule (no reproduction)
-Mule with Pai doesn't drain power (insert pai to Mule, Mule now will no lose power when in move)
-Cutting drive motor wires in Mule with Pai won't change the speed of Mule (insert pai to Mule, open Mule with screwdriver and cut drive motor wire with wirecutters, close Mule and turn Mule on)
-Pai Mules with both cut drive motor wires still can move (insert Pai to Mule, open it with screwdriver, cut both drive motor wires, close Mule and turn it on)

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This fixes #52281

- Player controlled mulebots now have their speed set based on the status of the two motor wires.  
- Player controlled mulebots will be immobilized if both their motor wires are cut.
- Mulebots no longer turn off power when the panel is opened with a screwdriver.

## Why It's Good For The Game

This makes persistence consistent.  Wires that affect the mulebot should also affect the player and not be ignored.    

## Changelog
:cl:
qol: Mulebots no longer turn off power when the panel is opened with a screwdriver.
fix: Player controlled mulebot mobility is now affected by motor wires. 
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
If you're looking for more mulebot bugs to fix, https://github.com/tgstation/tgstation/issues/59166 still exists, IIRC.

File: code/datums/wires/mulebot.dm [hunk 1]
```diff
@@ -19,6 +23,22 @@
 	if(M.open)
 		return TRUE
 
+/datum/wires/mulebot/on_cut(wire, mend)
+	var/mob/living/simple_animal/bot/mulebot/M = holder
```
Comment (Mothblocks) [inline code/datums/wires/mulebot.dm:27]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 2]
```diff
@@ -558,6 +556,8 @@
 		return
 	if(mode == BOT_IDLE)
 		return
+	if(HAS_TRAIT(src, TRAIT_IMMOBILIZED))
+		return
 
 	var/speed = (wires.is_cut(WIRE_MOTOR1) ? 0 : 1) + (wires.is_cut(WIRE_MOTOR2) ? 0 : 2)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm:562]: It looks like this is still doing its own work, despite the varspeed being set? Do you still need this?

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by timothymtorres** [comment]
> If you're looking for more mulebot bugs to fix, #59166 still exists, IIRC.

I was planning on knocking this out today.  It should be a small fix since I'm pretty sure I just need to delete a few lines.

File: code/modules/mob/living/simple_animal/bot/mulebot.dm (hunk 2 above)
Comment (timothymtorres) [reply code/modules/mob/living/simple_animal/bot/mulebot.dm:562]: This disables movement if both motor wires have been cut or if the mulebot has been turned off.

File: code/datums/wires/mulebot.dm (hunk 1 above)
Comment (timothymtorres) [reply code/datums/wires/mulebot.dm:27]: I can change it if you want but just be aware of the following:

- The entire file uses M to refer to the mulebot
- There is no other mob type involved in this file
- The file is really short

Given these reasons, I tried to keep it consistent.  I don't mind renaming all the `M` references if you prefer.

Comment (Mothblocks) [reply code/datums/wires/mulebot.dm:27]: I'm aware of it, yes--it would be okay to change all references, but not necessary other than this one.

Comment (timothymtorres) [reply code/datums/wires/mulebot.dm]: Fixed with https://github.com/tgstation/tgstation/pull/60909/commits/403aab48d6685a55432fa82ab6abb60c8896826f

**Pull request merged**